packages/components/src/components/hds/advanced-table/index.hbs

```diff
@@ -0,0 +1,76 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+
+<div
+  class={{this.classNames}}
+  ...attributes
+  role="grid"
+  aria-describedby={{this._captionId}}
+  {{style grid-template-columns=this.gridTemplateColumns}}
+>
+  <div id={{this._captionId}} class="sr-only hds-advanced-table__caption" aria-live="polite">
+    {{@caption}}
+    {{this.sortedMessageText}}
+  </div>
+  <div class="hds-advanced-table__thead" role="rowgroup">
+    <Hds::AdvancedTable::Tr
+      @selectionScope="col"
+      @onClickSortBySelected={{if @selectableColumnKey (fn this.setSortBy @selectableColumnKey)}}
+      @sortBySelectedOrder={{if (eq this._sortBy @selectableColumnKey) this._sortOrder}}
+      @isSelectable={{this.isSelectable}}
+      @onSelectionChange={{this.onSelectionAllChange}}
+      @didInsertCheckbox={{this.didInsertSelectAllCheckbox}}
+      @willDestroy={{this.willDestroySelectAllCheckbox}}
+      @selectionAriaLabelSuffix="all rows"
+    >
+      {{#each @columns as |column|}}
+        {{#if column.isSortable}}
+          <Hds::AdvancedTable::ThSort
+            @sortOrder={{if (eq column.key this._sortBy) this._sortOrder}}
+            @onClickSort={{fn this.setSortBy column.key}}
+            @align={{column.align}}
+            @tooltip={{column.tooltip}}
+          >
+            {{column.label}}
+          </Hds::AdvancedTable::ThSort>
+        {{else}}
+          <Hds::AdvancedTable::Th
+            @align={{column.align}}
+            @tooltip={{column.tooltip}}
+            @isVisuallyHidden={{column.isVisuallyHidden}}
+          >{{column.label}}</Hds::AdvancedTable::Th>
+        {{/if}}
+      {{/each}}
+    </Hds::AdvancedTable::Tr>
+  </div>
+
+  <div class="hds-advanced-table__tbody" role="rowgroup">
+    {{! ----------------------------------------------------------------------------------------
+        IMPORTANT: we loop on the `model` array and for each record
+        we yield the Tr/Td/Th elements _and_ the record itself as `data`
+        this means the consumer will *have to* use the `data` key to access it in their template
```

Comment:
question - What do you think about linking to an example in the docs for how this works?